### ISSUE TITLE
DotVVM.Tracing.ApplicationInsights

### ISSUE THREAD

**Issue by tomasherceg** [riganti/dotvvm#370]
We should create a new Nuget package which reports exceptions and request information data to the Application Insights. 

It would be registered in the `UseDotvvm` method, for example this way:

```
var dotvvmConfiguration = app.UseDotVVM<DotvvmStartup>(applicationPhysicalPath, options: options =>
{
    options.AddApplicationInsightsTracing()
});
```

1) This would register an implementation of `IRequestTracingReporter` in the `DotvvmConfiguration.Runtime.Tracing.Reporters` collection which we can add. 

When the requst ends, DotVVM should call the `TraceRequestAsync(context)` method on all configured reporters. 

2) It would also report all exceptions from commands or requests which occured during the request execution.

**Comment by tomasherceg** [comment]
@Rumbeer 
I like the implementation, but we should do several modifications to this:

In **OWIN**, the registration is simple: 
`options.AddApplicationInsightsTracing();`

In **ASP.NET Core**, the `TelemetryClient` is registered separately, and the reporter is registered in the `Configure` phase,because we don't have the `DotvvmConfiguration` object in the `ConfigureServices` method yet.
I think that we can add the `options.AddApplicationInsightsTracing` method to ASP.NET Core too. It should only register the `TelemetryClient` and `ApplicationInsightsReporter` in the `IServiceProvider`.
The `UseDotVVM` method shoul look for all implementations of `IRequestTracingReporter` and add them to the `Reporters` collection when the `DotvvmConfiguration` object is ready.

And finally, to make things consistent, I'd register the `ApplicationInsightsReporter` in DI in the OWIN version too.

**Comment by tomasherceg** [comment]
TBD

- [ ] Test it in several real projects.
- [ ] Check that the documentation is current.
- [ ] Check that is is merged to 1.1 too.

**Comment by quigamdev** [comment]
Already solved. The implementation was moved to https://github.com/riganti/dotvvm-tracing
